src/web_service/cookie.cpp

```diff
@@ -0,0 +1,1051 @@
+// SPDX-FileCopyrightText: 2023 Yuzu Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include <sstream>
+
+#include "common/string_util.h"
+#include "cookie.h"
+
+namespace WebService {
+
+#define TM_YEAR_BASE 1900
+
+Cookie::Cookie(std::string& name, std::string& value) : m_name{name}, m_value{value} {}
+
+bool Cookie::isSecure() const {
+    return m_secure;
+}
+
+void Cookie::setSecure(bool enable) {
+    m_secure = enable;
+}
+
+bool Cookie::isHttpOnly() const {
+    return m_httpOnly;
+}
+
+void Cookie::setHttpOnly(bool enable) {
+    m_httpOnly = enable;
+}
+
+Cookie::SameSite Cookie::sameSitePolicy() const {
+    return m_sameSite;
+}
+
+void Cookie::setSameSitePolicy(Cookie::SameSite sameSite_) {
+    m_sameSite = sameSite_;
+}
+
+std::tm Cookie::expirationDate() const {
+    return m_expirationDate;
+}
+
+void Cookie::setExpirationDate(const std::tm& date) {
+    m_expirationDate = date;
+}
+
+std::string Cookie::domain() const {
+    return m_domain;
+}
+
+void Cookie::setDomain(const std::string& domain_) {
```

Comment:
```suggestion
void Cookie::setDomain(std::string domain_) {
```

and std::move into the member variable